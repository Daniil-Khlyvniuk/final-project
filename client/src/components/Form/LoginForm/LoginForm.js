import React, { useState } from 'react'
import { useFormStyle } from '../../../utils/customHooks/useFormStyle'
import { Field, Form, Formik } from 'formik'
import { LOGIN_SCHEMA } from '../setting/Schemes'
import { Box, Typography, Button } from '@mui/material'
import { Link } from 'react-router-dom'
import CustomInput from '../setting/customElements/CustomInput'
import CustomSwitch from '../setting/customElements/CustomSwitch'

import useAuth from '../../../utils/customHooks/useAuth'


const LoginForm = () => {
	const classes = useFormStyle()
	const [serverResult, setServerResult] = useState(null)
	const {login} = useAuth()

	return (
		<Formik 
			initialValues = {{
				loginOrEmail: '',
				password: '',
				rememberMe: false,
			}}
			validationSchema = {LOGIN_SCHEMA}

			onSubmit = {async values => {
				try {
					const res = await login(values)
					if(res)
					{
						setServerResult({ success: 'You successfully Logged In' })
					}
				}
				catch (err) {
					setServerResult({ error: 'wrong login or password' })
				}
			}}
		>
			{(formikProps) => {
				return (
					<Form noValidate
						onSubmit={formikProps.handleSubmit}
					>
						<Field
							data-testid="loginOrEmail"
							component={CustomInput}
							label="Login or Email"
							name="loginOrEmail"
							type="text"
						/>
						<Field
							data-testid="password"
							component={CustomInput}
							label="Password"
							name="password"
							type="password"
						/>
						<Box 
							sx={{
								marginTop: '25px',
								paddingLeft: '15px',
							}}
						>
							<CustomSwitch 
								data-testid="rememberMe"
								name="rememberMe"
								label={
									<Typography
										component={'span'}
										sx={{
											fontSize: '14px',
											fontWeight: 300,
											lineHeight: '20px',
										}}
									>Remember me</Typography>
								}
								styles={{
									'& .MuiSwitch-switchBase': {
										'&.Mui-checked': {
											color:'#6FB7AC',
											'& + .MuiSwitch-track': {
												backgroundColor: '#6FB7AC',
											}
										},
									}}
								}
							/>
						</Box>

						<p className={classes.policy}>By signing up you agree to
							<Link to="/info/terms-of-service"> Terms of Service </Link> 
							and <Link to="/info/privacy-policy"> Privacy Policy </Link>
						</p>

						{serverResult && serverResult.error && (
							<Box className={classes.formStatusBlock}>
								<p className={classes.error}>{serverResult.error}</p>
							</Box>
						)}

						{serverResult && serverResult.success && (
							<Box className={classes.formStatusBlock}>
								<p className={classes.success}>{serverResult.success}</p>
							</Box>
						)}

						<Box sx={{ display: 'flex', justifyContent: 'center' }}>
							<Button
								type='submit'
								variant="contained"
								direction="form"
								disabled={
									!formikProps.isValid ||
									formikProps.isSubmitting
								}
							>
								log in
							</Button>
						</Box>
					</Form>
				)
			}}
		</Formik>
	)
}

export default LoginForm