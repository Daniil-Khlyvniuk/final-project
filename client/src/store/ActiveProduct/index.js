
export { default as activeProductSelector } from './selectors'
export { default as activeProductOperations } from './operations'
import activeProductReducer from './activeProductSlice'

export default activeProductReducer