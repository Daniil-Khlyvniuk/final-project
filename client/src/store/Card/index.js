export { default as cardSelectors } from './selectors'
export { default as cardOperations } from './operations'
import * as cardActions from './cardSlice'

export default cardActions


