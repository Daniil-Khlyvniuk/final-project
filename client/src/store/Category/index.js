export { default as categorySelectors } from './selectors'
export { default as categoryOperations } from './operations'

import * as categoryActions from './categorySlice'
export default categoryActions