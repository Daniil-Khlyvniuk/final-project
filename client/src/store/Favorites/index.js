export { default as favoritesSelectors } from './selectors'
export { default as favoritesOperations } from './operations'

import * as favoritesActions from './favoritesSlice'
export default favoritesActions