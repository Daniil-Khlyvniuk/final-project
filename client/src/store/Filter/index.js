import reducer from './filterSlice'

export {default as filterSelectors} from './selectors'
export {default as filterOperations} from './operations'

export default reducer