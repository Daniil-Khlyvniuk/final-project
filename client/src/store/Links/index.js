export {default as linksSelectors} from './selectors'
export {default as linksOperations} from './operations'

import linksReducer from './linksSlice'
export default linksReducer