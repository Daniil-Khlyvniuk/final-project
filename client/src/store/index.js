import { configureStore } from '@reduxjs/toolkit'
import productsSlice from './Products/productsSlice'
import linksSlice from './Links/'
import categorySlice from './Category/categorySlice'
import sliderSlice from './Slider'
import modalSlice from './Modal/modalSlice'
import shoppingBagSlice from './ShoppingBag/shoppingBagSlice'
import productSlice from './Product/productSlice'
import filterSlice from './Filter'

import userSlice from './User'

const store = configureStore({
	reducer: {
		slides: sliderSlice,
		products: productsSlice,
		links: linksSlice,
		category: categorySlice,
		modal: modalSlice,
		shoppingBag: shoppingBagSlice,
		user: userSlice,
		product : productSlice,
		filter: filterSlice,

	},
	middleware: (getDefaultMiddleware) =>
		getDefaultMiddleware({
			serializableCheck: {
				// Ignore these action types
				ignoredActions: ['modal/modalToggle', 'user/setToken', 'user/fetchUser/pending','user/fetchUser/fulfilled','filter/setFiltersFromQueryString'],
			},
		}),
})

export default store
