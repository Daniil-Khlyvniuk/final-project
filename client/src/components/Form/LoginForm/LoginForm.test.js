import React from 'react'
import LoginForm from './LoginForm'
import { BrowserRouter } from 'react-router-dom'
import {Provider} from 'react-redux'
import store from '../../../store'
import {render, screen, waitFor} from '@testing-library/react'
import userEvent from '@testing-library/user-event'

describe('Smoke test LoginFrom', () => {
	const setUp = () => {
		render(
			<Provider store={store}>
				<BrowserRouter>
					<LoginForm/>
				</BrowserRouter>
			</Provider>
		)
	}

	let container
	beforeEach(() => {
		container = setUp()
	})
	it('should render LoginForm', () => {
		expect(container)
	})
	it('LoginForm Snapshot test', () => {
		expect(container).toMatchSnapshot()
	})
})

describe("Testing formik on LoginForm", () => {
	it('should rendering and submitting a basic Formik form',  () => {
		const handleSubmit = jest.fn()

		render(
			<Provider store={store}>
				<BrowserRouter>
					<LoginForm/>
				</BrowserRouter>
			</Provider>
		)

		userEvent.type(screen.getByLabelText(/Login or Emai/i), 'john.dee@someemail.com')
		userEvent.type(screen.getByLabelText(/Password/i), 'qwerty1234')

		userEvent.click(screen.getByTestId('button', {type: /submit/i}))

		 waitFor(() =>
			expect(handleSubmit).toHaveBeenCalledWith({
				email: 'john.dee@someemail.com',
				password: 'qwerty1234',
			}),
		)
	})
})