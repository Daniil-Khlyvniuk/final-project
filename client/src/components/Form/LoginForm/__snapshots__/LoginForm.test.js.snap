// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Smoke test LoginFrom LoginForm Snapshot test 1`] = `undefined`;
